gnomad_qc/v3/create_release/create_hgdp_tgp_subset.py

```diff
@@ -563,26 +563,28 @@ def create_full_subset_dense_mt(
     logger.info("Splitting multi-allelics...")
     mt = hl.experimental.sparse_split_multi(mt, filter_changed_loci=True)
 
-    logger.info("Computing adj and sex adjusted genotypes...")
-    mt = mt.annotate_entries(
-        GT=adjusted_sex_ploidy_expr(
-            mt.locus, mt.GT, mt.gnomad_sex_imputation.sex_karyotype
-        ),
-        adj=get_adj_expr(mt.GT, mt.GQ, mt.DP, mt.AD),
-    )
-
     logger.info(
         "Setting het genotypes at sites with > 1% AF (using precomputed v3.0"
         " frequencies) and > 0.9 AB to homalt..."
     )
     # NOTE: Using v3.0 frequencies here and not v3.1 frequencies because the frequency code adjusted genotypes (homalt depletion fix) using v3.0 frequencies # noqa
     # https://github.com/broadinstitute/gnomad_qc/blob/efea6851a421f4bc66b73db588c0eeeb7cd27539/gnomad_qc/v3/annotations/generate_freq_data_hgdp_tgp.py#L129
+    mt = mt.annotate_entries(unadjusted_GT=mt.GT)
     freq_ht = release_sites(public=True).versions["3.0"].ht().select("freq")
     mt = hom_alt_depletion_fix(
         mt, het_non_ref_expr=mt._het_non_ref, af_expr=freq_ht[mt.row_key].freq[0].AF
     )
     mt = mt.drop("_het_non_ref")
 
+    logger.info("Computing adj and sex adjusted genotypes...")
```

Comment:
I think this code is in the correct order -- it applies the homalt hotfix first, which adjusts high AB het GTs from XY samples on chrX/chrY to homozygous GTs. This means those GTs get set to haploid when adjusting for [adj](https://github.com/broadinstitute/gnomad_methods/blob/main/gnomad/utils/annotations.py#L605) rather than None, which means we count those sample GTs towards both the AC and the AN.

Checking v4, though, I think we [adjusted sex ploidy](https://github.com/broadinstitute/gnomad_methods/blob/main/gnomad/sample_qc/sex.py#L18) before applying the hotfix based on this [line](https://github.com/broadinstitute/gnomad_qc/blob/main/gnomad_qc/v4/annotations/generate_freq.py#L535) -- is that accurate?

Just wanted to flag this because I'm not 100% clear how this MT is getting used to to create the `pop_diff` resource Qin mentioned here https://the-tgg.slack.com/archives/C018PB68VU5/p1698336165273119?thread_ts=1698335729.900479&cid=C018PB68VU5

Reply:
just noting here that chatted about this in a v4 sync -- takeaways were

- We are going to keep this order (applying homalt hotfix first)
- We are going to keep multiple versions of the GTs: the original, unadjusted GT; the fixed GT with this rerun, and the version of the GT that would have been created with the v3.1 code